### ISSUE TITLE
produce: certificate verify failed on travis-ci with ruby 2.1.5

### ISSUE THREAD

**Issue by Sega-Zero** [fastlane/fastlane#9042]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

Trying to build my app on a travis instance with `rvm: 2.1.5`.
The build fails at the produce part with an error: `SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed`
Is ruby 2.1.5 not supported by fastlane? What's the recommended ruby version (cannot use 2.0.0 version, since nokogiri requires at least 2.1.0)

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

```shell
[19:52:45]: ---------------------
[19:52:45]: --- Step: produce ---
[19:52:45]: ---------------------
+----------------+----------------------------------+
|            Summary for produce 2.28.6             |
+----------------+----------------------------------+
| app_identifier | <REDACTED> |
| app_name       | <REDACTED>   |
| skip_itc       | true                             |
| username       | travis@mail.com                 |
| sku            | <REDACTED>                       |
| platform       | ios                              |
| language       | English                          |
| skip_devcenter | false                            |
| team_id        | <REDACTED>                       |
| team_name      | <REDACTED>       |
| itc_team_id    | <REDACTED>                         |
+----------------+----------------------------------+
[19:52:45]: -------------------
[19:52:45]: --- Step: is_ci ---
[19:52:45]: -------------------
[19:52:45]: --------------------------
[19:52:45]: --- Step: fastlane env ---
[19:52:45]: --------------------------
+------------------+----------+
|        Lane Context         |
+------------------+----------+
| DEFAULT_PLATFORM | ios      |
| PLATFORM_NAME    | ios      |
| LANE_NAME        | ios auto |
+------------------+----------+
[19:52:48]: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
+------+---------------------------------------------------------------+-------------+
|                                  fastlane summary                                  |
+------+---------------------------------------------------------------+-------------+
| Step | Action                                                        | Time (in s) |
+------+---------------------------------------------------------------+-------------+
| 1    | default_platform                                              | 0           |
| 2    | git_branch                                                    | 0           |
| 3    | create_keychain                                               | 0           |
| 4    | sed -i '' "s#\(Pods\.framework in Frameworks \*/ = {isa = PBX | 0           |
| 5   | Switch to ios beta lane                                       | 0           |
| 6   | is_ci                                                         | 0           |
| 7   | update_app_identifier                                         | 0           |
| 💥   | produce                                                       | 0           |
| 9   | is_ci                                                         | 0           |
| 10   | fastlane env                                                  | 2           |
+------+---------------------------------------------------------------+-------------+
[19:52:48]: fastlane finished with errors
[19:52:48]: -----------------------------------------------------------------------
[19:52:48]: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
[19:52:48]: 
[19:52:48]: SSL errors can be caused by various components on your local machine.
[19:52:48]: 
[19:52:48]: The best solution is to use the self-contained fastlane version.
[19:52:48]: Which ships with a bundled OpenSSL,ruby and all gems - so you don't depend on system libraries
[19:52:48]:  - Use One-Click-Installer:
[19:52:48]:     - download fastlane at https://download.fastlane.tools
[19:52:48]: -----------------------------------------------------------
[19:52:48]:     - extract the archive and double click the `install`
[19:52:48]: -----------------------------------------------------------
[19:52:48]:  - Use Homebrew
[19:52:48]:     - update brew with `brew update`
[19:52:48]:     - install fastlane:
[19:52:48]: -----------------------------------------------------------
[19:52:48]:       - 🚀 `brew cask install fastlane` 🚀
[19:52:48]: -----------------------------------------------------------
[19:52:48]: for more details on ways to install fastlane please refer the documentation:
[19:52:48]: -----------------------------------------------------------
[19:52:48]:         🚀       https://docs.fastlane.tools          🚀   
[19:52:48]: -----------------------------------------------------------
[19:52:48]: 
[19:52:48]: You can also install a new version of Ruby
[19:52:48]: 
[19:52:48]: - Make sure OpenSSL is installed with Homebrew: `brew update && brew upgrade openssl`
[19:52:48]: - If you use system Ruby:
[19:52:48]:   - Run `brew update && brew install ruby`
[19:52:48]: - If you use rbenv with ruby-build:
[19:52:48]:   - Run `brew update && brew upgrade ruby-build && rbenv install 2.3.1`
[19:52:48]:   - Run `rbenv global 2.3.1` to make it the new global default Ruby version
[19:52:48]: - If you use rvm:
[19:52:48]:   - First run `rvm osx-ssl-certs update all`
[19:52:48]:   - Then run `rvm reinstall ruby-2.3.1 --with-openssl-dir=/usr/local`
[19:52:48]: 
[19:52:48]: If that doesn't fix your issue, please google for the following error message:
[19:52:48]:   'SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed'
[19:52:48]: -----------------------------------------------------------------------
[!] SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
```

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

[19:52:45]: $ fastlane env
[19:52:46]: ▸ /Users/travis/build/company/iOS/vendor/bundle/ruby/2.1.0/gems/fastimage-2.1.0/lib/fastimage.rb:67: warning: already initialized constant URI::DEFAULT_PARSER
[19:52:46]: ▸ /Users/travis/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/uri/common.rb:545: warning: previous definition of DEFAULT_PARSER was here
[19:52:46]: ▸ [19:52:46]: Generating fastlane environment output, this might take a few seconds...
[19:52:47]: ▸ <details><summary>✅ fastlane environment ✅</summary>
[19:52:47]: ▸ ### Stack
[19:52:47]: ▸ | Key                         | Value                                                      |
[19:52:47]: ▸ | --------------------------- | ---------------------------------------------------------- |
[19:52:47]: ▸ | OS                          | 10.12.1                                                    |
[19:52:47]: ▸ | Ruby                        | 2.1.5                                                      |
[19:52:47]: ▸ | Bundler?                    | true                                                       |
[19:52:47]: ▸ | Git                         | git version 2.11.0                                         |
[19:52:47]: ▸ | Installation Source         | ~/build/company/iOS/vendor/bundle/ruby/2.1.0/bin/fastlane |
[19:52:47]: ▸ | Host                        | Mac OS X 10.12.1 (16B2555)                                 |
[19:52:47]: ▸ | Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.1.5/lib                               |
[19:52:47]: ▸ | OpenSSL Version             | OpenSSL 1.0.1j 15 Oct 2014                                 |
[19:52:47]: ▸ | Is contained                | false                                                      |
[19:52:47]: ▸ | Is homebrew                 | false                                                      |
[19:52:47]: ▸ | Is installed via Fabric.app | false                                                      |
[19:52:47]: ▸ | Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                |
[19:52:47]: ▸ | Xcode Version               | 8.2                                                        |
[19:52:47]: ▸ ### System Locale
[19:52:47]: ▸ | Variable | Value       |   |
[19:52:47]: ▸ | -------- | ----------- | - |
[19:52:47]: ▸ | LANG     | en_US.UTF-8 | ✅ |
[19:52:47]: ▸ | LC_ALL   | en_US.UTF-8 | ✅ |
[19:52:47]: ▸ | LANGUAGE |             |   |
<details><summary>`./Appfile`</summary>
[19:52:47]: ▸ ```ruby
[19:52:47]: ▸ app_identifier "<REDACTED>"
[19:52:47]: ▸ apple_id "travis@mail.com"
[19:52:47]: ▸ itc_team_id "<REDACTED>"
[19:52:47]: ▸ team_id "<REDACTED>"
[19:52:47]: ▸ team_name "<REDACTED>"```
[19:52:47]: ▸ </details>
[19:52:47]: ▸ ### fastlane gems
[19:52:47]: ▸ | Gem      | Version | Update-Status |
[19:52:47]: ▸ | -------- | ------- | ------------- |
[19:52:47]: ▸ | fastlane | 2.28.6  | ✅ Up-To-Date  |
[19:52:47]: ▸ ### Loaded fastlane plugins:
[19:52:47]: ▸ **No plugins Loaded**
[19:52:47]: ▸ <details><summary><b>Loaded gems</b></summary>
[19:52:47]: ▸ | Gem                       | Version      |
[19:52:47]: ▸ | ------------------------- | ------------ |
[19:52:47]: ▸ | CFPropertyList            | 2.3.5        |
[19:52:47]: ▸ | i18n                      | 0.8.1        |
[19:52:47]: ▸ | minitest                  | 5.10.1       |
[19:52:47]: ▸ | thread_safe               | 0.3.6        |
[19:52:47]: ▸ | tzinfo                    | 1.2.3        |
[19:52:47]: ▸ | activesupport             | 4.2.8        |
[19:52:47]: ▸ | public_suffix             | 2.0.5        |
[19:52:47]: ▸ | addressable               | 2.5.1        |
[19:52:47]: ▸ | babosa                    | 1.0.2        |
[19:52:47]: ▸ | bundler                   | 1.14.6       |
[19:52:47]: ▸ | claide                    | 1.0.1        |
[19:52:47]: ▸ | colored                   | 1.2          |
[19:52:47]: ▸ | colored2                  | 3.1.2        |
[19:52:47]: ▸ | highline                  | 1.7.8        |
[19:52:47]: ▸ | commander-fastlane        | 4.4.4        |
[19:52:47]: ▸ | unf_ext                   | 0.0.7.4      |
[19:52:47]: ▸ | unf                       | 0.1.4        |
[19:52:47]: ▸ | domain_name               | 0.5.20170404 |
[19:52:47]: ▸ | dotenv                    | 2.2.1        |
[19:52:47]: ▸ | excon                     | 0.55.0       |
[19:52:47]: ▸ | multipart-post            | 2.0.0        |
[19:52:47]: ▸ | faraday                   | 0.12.1       |
[19:52:47]: ▸ | http-cookie               | 1.0.3        |
[19:52:47]: ▸ | faraday-cookie_jar        | 0.0.6        |
[19:52:47]: ▸ | faraday_middleware        | 0.11.0.1     |
[19:52:47]: ▸ | fastimage                 | 2.1.0        |
[19:52:47]: ▸ | gh_inspector              | 1.0.3        |
[19:52:47]: ▸ | jwt                       | 1.5.6        |
[19:52:47]: ▸ | little-plugger            | 1.1.4        |
[19:52:47]: ▸ | multi_json                | 1.12.1       |
[19:52:47]: ▸ | logging                   | 2.2.2        |
[19:52:47]: ▸ | memoist                   | 0.15.0       |
[19:52:47]: ▸ | os                        | 0.9.6        |
[19:52:47]: ▸ | signet                    | 0.7.3        |
[19:52:47]: ▸ | googleauth                | 0.5.1        |
[19:52:47]: ▸ | httpclient                | 2.8.3        |
[19:52:47]: ▸ | hurley                    | 0.2          |
[19:52:47]: ▸ | mime-types-data           | 3.2016.0521  |
[19:52:47]: ▸ | mime-types                | 3.1          |
[19:52:47]: ▸ | uber                      | 0.0.15       |
[19:52:47]: ▸ | representable             | 2.3.0        |
[19:52:47]: ▸ | retriable                 | 2.1.0        |
[19:52:47]: ▸ | google-api-client         | 0.9.28       |
[19:52:47]: ▸ | json                      | 2.1.0        |
[19:52:47]: ▸ | mini_magick               | 4.5.1        |
[19:52:47]: ▸ | multi_xml                 | 0.6.0        |
[19:52:47]: ▸ | plist                     | 3.3.0        |
[19:52:47]: ▸ | rubyzip                   | 1.2.1        |
[19:52:47]: ▸ | security                  | 0.1.3        |
[19:52:47]: ▸ | slack-notifier            | 1.5.1        |
[19:52:47]: ▸ | terminal-notifier         | 1.7.1        |
[19:52:47]: ▸ | unicode-display_width     | 1.1.3        |
[19:52:47]: ▸ | terminal-table            | 1.7.3        |
[19:52:47]: ▸ | tty-screen                | 0.5.0        |
[19:52:47]: ▸ | word_wrap                 | 1.0.0        |
[19:52:47]: ▸ | nanaimo                   | 0.2.3        |
[19:52:47]: ▸ | xcodeproj                 | 1.4.4        |
[19:52:47]: ▸ | rouge                     | 2.0.7        |
[19:52:47]: ▸ | xcpretty                  | 0.2.7        |
[19:52:47]: ▸ | xcpretty-travis-formatter | 0.0.4        |
[19:52:47]: ▸ | mustache                  | 0.99.8       |
[19:52:47]: ▸ | mini_portile2             | 2.1.0        |
[19:52:47]: ▸ | nokogiri                  | 1.7.1        |
[19:52:47]: ▸ | redcarpet                 | 3.4.0        |
[19:52:47]: ▸ | sass                      | 3.4.23       |
[19:52:47]: ▸ | jazzy                     | 0.0.14       |
[19:52:47]: ▸ </details>
[19:52:47]: ▸ *generated on:* **2017-04-28**
[19:52:47]: ▸ </details>

**Comment by KrauseFx** [comment]
```
[19:52:48]: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed
[19:52:48]: 
[19:52:48]: SSL errors can be caused by various components on your local machine.
[19:52:48]: 
[19:52:48]: The best solution is to use the self-contained fastlane version.
[19:52:48]: Which ships with a bundled OpenSSL,ruby and all gems - so you don't depend on system libraries
[19:52:48]:  - Use One-Click-Installer:
[19:52:48]:     - download fastlane at https://download.fastlane.tools
[19:52:48]: -----------------------------------------------------------
[19:52:48]:     - extract the archive and double click the `install`
[19:52:48]: -----------------------------------------------------------
[19:52:48]:  - Use Homebrew
[19:52:48]:     - update brew with `brew update`
[19:52:48]:     - install fastlane:
[19:52:48]: -----------------------------------------------------------
[19:52:48]:       - 🚀 `brew cask install fastlane` 🚀
[19:52:48]: -----------------------------------------------------------
[19:52:48]: for more details on ways to install fastlane please refer the documentation:
[19:52:48]: -----------------------------------------------------------
[19:52:48]:         🚀       https://docs.fastlane.tools          🚀   
[19:52:48]: -----------------------------------------------------------
[19:52:48]: 
[19:52:48]: You can also install a new version of Ruby
[19:52:48]: 
[19:52:48]: - Make sure OpenSSL is installed with Homebrew: `brew update && brew upgrade openssl`
[19:52:48]: - If you use system Ruby:
[19:52:48]:   - Run `brew update && brew install ruby`
[19:52:48]: - If you use rbenv with ruby-build:
[19:52:48]:   - Run `brew update && brew upgrade ruby-build && rbenv install 2.3.1`
[19:52:48]:   - Run `rbenv global 2.3.1` to make it the new global default Ruby version
[19:52:48]: - If you use rvm:
[19:52:48]:   - First run `rvm osx-ssl-certs update all`
[19:52:48]:   - Then run `rvm reinstall ruby-2.3.1 --with-openssl-dir=/usr/local`
[19:52:48]: 
[19:52:48]: If that doesn't fix your issue, please google for the following error message:
[19:52:48]:   'SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed'
```

There is the list of our recommended ways to solve the issue. Basically you have to make sure to use a Ruby with an up to date OpenSSL version 👍

**Comment by Sega-Zero** [comment]
solved an issue by specifying rvm: 2.3.1 in .travis.yml
